contrib/olivere/elastic/elastictrace.go

```diff
@@ -117,25 +118,21 @@ func peek(rc io.ReadCloser, gzipped bool, max, n int) (string, io.ReadCloser, er
 	}
 	snip, err := r.Peek(n)
 	if err == io.EOF {
-		err = nil
+		return string(snip), rc2, nil
 	}
 
-	// the snippet we just pulled out may be gzip encoded. if so, we need to
-	// decode it before we return so we can report the actual text to DataDog
-	if gzipped {
-		reader, err := gzip.NewReader(bytes.NewReader(snip))
+	if encoding == "gzip" {
+		// unpack the snippet
+		gzr, err := gzip.NewReader(bytes.NewReader(snip))
 		if err != nil {
-			return string(snip), rc2, err
+			return string(snip), rc2, nil
```

Comment:
Given that gzipped data is effectively never useful, I think these two error conditions should return the `err` instead of a `nil` so that the data isn't used in the caller. I had them returning `nil` since the `EOF` above did that. Thoughts?

Reply:
I think what you have now is fine. The gzip reader will only return an error if it tries to [read the gzip header](https://golang.org/src/compress/gzip/gunzip.go?s=2831:2875#L172) and it can't. If that's the case, we can imply that the content is not gzip, so I guess the best outcome is to just return the content as it is, whatever it is. It may be more useful than the fallback which simply uses the status code, on error.

Because of the comment you made, maybe it's worthwhile putting a comment inside the `if` condition in the same style as we did the other one, something like:
```go
if err != nil {
    // snip wasn't gzip; return it as is
```
Or something along these lines.